Format test vectors and include them in the draft (#67)
cfrg/draft-irtf-cfrg-frost: draft-irtf-cfrg-frost.md

    the output to a Scalar as described in {{!RISTRETTO, Section 4.4}}.
  - H3(m): Implemented by computing H(contextString \|\| "digest" \|\| m).

## FROST(edwards448, SHAKE256)

This ciphersuite uses edwards448 for the Group and SHA256 for the Hash function `H`

> meant to produce signatures indistinguishable from Ed448 as specified in {{!RFC8032}}.

Author: ```suggestion
meant to produce signatures compatible with Ed448 as specified in {{!RFC8032}}.
```

constituent parts. The input message to be signed is also encoded as a hexadecimal
string.

Each test vector consists of the following information.

- Configuration: This lists the fixed parameters for the particular instantiation
  of FROST, including MAX_SIGNERS, THRESHOLD_LIMIT, and NUM_SIGNERS.
- Group input parameters: This lists the group secret key and shared public key,
  generated by a trusted dealer as described in {{dep-dealer}}, as well as the
  input message to be signed. All values are encoded as hexadecimal strings.
- Signer input parameters: This lists the signing key share for each of the
  NUM_SIGNERS signers.
- Round one parameters and outputs: This lists the NUM_SIGNERS participants engaged
  in the protocol, identified by their integer index, the hiding and binding commitment
  values produced in {{frost-round-one}}, as well as the resulting group commitment list,
  encoded as described in {{dep-encoding}}, and group binding factor as computed in {{frost-round-two}}).
- Round two parameters and outputs: This lists the NUM_SIGNERS participants engaged
  in the protocol, identified by their integer index, along with their corresponding

>   output signature and commitment share as produced in {{frost-round-two}}.

Reviewer: ```suggestion
  output signature share and group commitment share as produced in {{frost-round-two}}.
```

## FROST(Ed25519, SHA512)

~~~
// Configuration information
MAX_SIGNERS: 3
THRESHOLD_LIMIT: 2
NUM_SIGNERS: 2

// Group input parameters
group_secret_key: 7c1c33d3f5291d85de664833beb1ad469f7fb6025a0ec78b3a7
90c6e13a98304
group_public_key: 377a6acb3b9b5f642c5ce355d23cac0568aad0da63c633d59d4
168bdcbce35af
message: 74657374

// Signer input parameters
S1 signer_share: 949dcc590407aae7d388761cddb0c0db6f5627aea8e217f4a033
f2ec83d93509
S2 signer_share: ac1e66e012e4364ac9aaa405fcafd370402d9859f7b6685c07ee
d76bf409e80d
S3 signer_share: d7cb090a075eb154e82fdb4b3cb507f110040905468bb9c46da8
bdea643a9a02

// Round one parameters
participants: 1,2
commitment_list: 000178e175d15cb5cec1257e0d84d797ba8c3dd9b4c7bc50f3fa
527c200bcc6c4a954cdad16ae67ac5919159d655b681bd038574383bab423614f8967
396ee12ca62000288a4e6c3d8353dc3f4aca2e10d10a75fb98d9fbea98981bfb25375
996c5767c932bbf10c41feb17d41cc6433e69f16cceccc42a00aedf72feb5f44929fd
f2e2fee26b0dd4af7e749aa1a8ee3c10ae9923f618980772e473f8819a5d4940e0db2
7ac185f8a0e1d5f84f88bc887fd67b143732c304cc5fa9ad8e6f57f50028a8ff
group_binding_factor: c4d7668d793ff4c6ec424fb493cdab3ef5b625eefffe775

> 71ff28a345e5f700a

Reviewer: Should this go after the Signer round one outputs?

Author: Eh, I don't feel strongly. It could.

- Group input parameters: This lists the group secret key and shared public key,
  generated by a trusted dealer as described in {{dep-dealer}}, as well as the
  input message to be signed. All values are encoded as hexadecimal strings.
- Signer input parameters: This lists the signing key share for each of the
  NUM_SIGNERS signers.
- Round one parameters and outputs: This lists the NUM_SIGNERS participants engaged
  in the protocol, identified by their integer index, the hiding and binding commitment
  values produced in {{frost-round-one}}, as well as the resulting group commitment list,
  encoded as described in {{dep-encoding}}, and group binding factor as computed in {{frost-round-two}}).
- Round two parameters and outputs: This lists the NUM_SIGNERS participants engaged
  in the protocol, identified by their integer index, along with their corresponding
  output signature share and group commitment share as produced in {{frost-round-two}}.
- Final output: This lists the aggregate signature as produced in {{frost-aggregation}}.

## FROST(Ed25519, SHA512)

~~~
// Configuration information
MAX_SIGNERS: 3
THRESHOLD_LIMIT: 2
NUM_SIGNERS: 2

// Group input parameters

> group_secret_key: 7c1c33d3f5291d85de664833beb1ad469f7fb6025a0ec78b3a7

Author: Right -- one can test that Shamir recovery yields this group secret.

4c116f22ca901
S1 hiding_nonce_commitment: 824e9eddddf02b2a9caf5859825e999d791ca094f
65b814a8bca6013d9cc3127
S1 binding_nonce_commitment: 74c7e1271d2939a84a9a867e3a06579b4d25659b
427439ccf0d745b43f75b766
S2 hiding_nonce: 4d66d319f20a728ec3d491cbf260cc6be687bd87cc2b5fdb4d5f
528f65fd650d
S2 binding_nonce: 278b9b1e04632e6af3f1a3c144d07922ffcf5efd3a341b47abc
19c43f48ce306
S2 hiding_nonce_commitment: 8013834ff4d48e7d6b76c2e732bc611f54720ef89
33c4ca4de7eaaa77ff5cd12
S2 binding_nonce_commitment: 5e056ecc4f7c4657d3a742354430d768f945db22
9c335d258e9622ad99f3e758

// Round two parameters
participants: 1,2

// Signer round two outputs
S1 sig_share: 503c943f6af38f3ad2271de3d792c4756c5d9c483989e2005add0e8
9b504dc01
S1 group_commitment_share: 40587ff15e9cba4fe601c4378b40ae4babe2ea6f0b
549a6712d84837166b954d
S2 sig_share: eee969c17354ec84933723b45ac9965d2874e21c3a50cd31648f708
bfb9f160b
S2 group_commitment_share: 70661957faf398410a9da3e20cfd6f1233b368548e
44ffccee3143a42b691844

sig: 90ac875023a311624948b2660bc5524f690ae9e14fcd6541959bee2e868d8c32
3e26fe00de477cbf655f4097325c5bd394d17e6573d9af32be6c7f14b1a4f20c
~~~

## FROST(P-256, SHA256)

~~~
// Configuration information

> MAX_SIGNERS: 3

Reviewer 2: Should we have a test for a different number of signers, just in case?

Author: Do you mean a different number for `MAX_SIGNERS`, or `NUM_SIGNERS`? I don't think changing `MAX_SIGNERS` increases test coverage. Changing `NUM_SIGNERS` might, however. I would prefer to do that in a separate PR.
